performance_tests/test_api_compatibility.py

```
@@ -0,0 +1,271 @@
+        # Create a test record
+        context = LogContext()
+        record = LogRecord(
+            level=LogLevel.INFO,
+            message="Test message",
+            context=context,
+            timestamp=1234567890.0
+        )
+        
+        # Process the record
+        result = pipeline.process(record)
+        assert result is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)